### ISSUE TITLE
Make document "id" field non-enumerable

### ISSUE THREAD

**Issue by Chaldron** [mesqueeb/vuex-easy-firestore#194]
I'm not sure if this is a bug with the way I've setup my store, but every document/collection has an "id" field which is equal to the Firestore document ID. I personally don't see the use case for including this field since the documents are indexed by this ID anyway, but even if it's included I think it would make more sense for it to be a non-enumerable property. This would make iterating over the fields of a document/collection easier as well, since you would only iterate over the known documents/fields and not the extra "id" field which isn't expected.

For example, I want to instantiate a component using props derived from the documents in my collection. However, I can't do something like the following:

`<Component v-for="elem in myModule.elems" myProp="elem.data" .../>`

Since an error will be thrown when we hit the "id" field.

Let me know if there's a better way around this, but I believe the vuefire/vuexfire libraries avoid this problem by using non-enumerable properties.

**Comment by mesqueeb** [comment]
Thanks for this issue!

Since at this point it would be a breaking change, I'll keep this in mind for v2.0! (planned for when Vuex 3.0 releases)

Feel free to make issues on any other suggestions as well! 🙃

<sub>--<br>Vuex Easy Firestore was made with ♥ by Luca Ban.<br>If this library helped you in any way you can support me by buying me <a href="https://www.buymeacoffee.com/mesqueeb">a cup of coffee</a>.</sub>

**Comment by louisameline** [comment]
I believe this issue can safely be requalified as `Stop including "id" in documents`.

**Comment by Chaldron** [comment]
> I believe this issue can safely be requalified as `Stop including "id" in documents`.

That's an option, but there's no need to necessarily remove it. The property could just be defined as non-enumerable.

**Comment by louisameline** [comment]
I see no reason to include it, unless the user specifically wants to. Do you?

**Comment by mesqueeb** [comment]
I realise now that adding the `id` to the actual document is forcing my own methods on all devs. This is something that _should_ be handled in hooks I'm thinking.
😆

**Comment by fergusmeiklejohn** [comment]
Just wanted to comment that I'm using the `id` you add, and for me it's a good idea to have it, so as removing it would be a breaking change can that be made less painful please?
thanks :-)

**Comment by mesqueeb** [comment]
@fergusmeiklejohn Yes of course, I won't remove this feature without being able to easily enable it again!!

Basically for version 2.0, I do believe that "by default" we should not add the id on the document, as it modifies the data in a way that might be unexpected for the developer, since he/she never explicitly did so.

However..... For me personally, I use this feature in all my databases, and it's so useful when a database enforces this paradigm, because you will have the id automatically usable on documents in the front-end.

So I will make it very easy to enable this feature again in version 2.0, just not by default 😉

**Comment by louisameline** [comment]
In most cases, the id will be the document name, and has no use to be repeated within the document. But of course you'll be able to include it if you want.
